src/auto-complete-element.js

```diff
@@ -92,5 +123,24 @@ export default class AutocompleteElement extends HTMLElement {
         )
         break
     }
+
+    setValidity(this)
+  }
+
+  formResetCallback() {
+    this.value = ''
+  }
+}
+
+function setValidity(el) {
+  const autocomplete = state.get(el)
+  if (!autocomplete) return
+  const internals = elementInternals.get(el)
+  if (!internals) return
+
+  if (el.required) {
+    internals.setValidity({valueMissing: !el.value}, 'Please select an option.', autocomplete.input)
```

Comment:
https://html.spec.whatwg.org/multipage/custom-elements.html#dom-elementinternals-setvalidity

The third argument sets a validation anchor, this would be the element that gets focused on when `reportValidity()` is called.